lib/identity/hostdata/settings.rb

```diff
@@ -0,0 +1,68 @@
+module Identity
+  module Hostdata
+    # Handles reading settings and configuration from a YAML file, allows
+    # overriding via ENV vars
+    class Settings
+      attr_reader :config
+
+      # @param [Hash] configuration
+      # @param [String] rails_env from +Rails.env+
+      def initialize(configuration:, rails_env:)
+        @config = {}
+
+        keys = Set.new(configuration.keys - %w[production development test])
+        keys |= configuration[rails_env]&.keys || []
+
+        keys.each do |key|
+          value = configuration.dig(rails_env, key) || configuration[key]
+          env_value = ENV[key]
+
+          check_string_key(key)
+          check_string_value(key, value)
+
+          if env_value
+            warn "WARNING: #{key} is being loaded from ENV instead of application.yml"
+            @config[key] = env_value
+          else
+            @config[key] = value
+            ENV[key] = value
```

Comment:
Can we drop this part? It was needed in the IDP because we used config.yml to set an ENV variable for a library, but I'm not sure if that's the case here.

Reply:
You mean just the setting part, not the overriding existing values part, right?

Would you be happier if we left it in as a boolean option (`write_to_env`)? The behavior is still needed in IDP for AAMVA, LexisNexis gems

Reply:
ohh, I misunderstood, the IDP will eventually inherit and use this behavior.

If non-IDP apps will also use it, yeah, I think it would be good to default to not writing to ENV.

Reply:
Updated in 55e524489f4472bb22920cfec622048d4818aa5b